### PR TITLE
Feat: timesheets collection

### PR DESCRIPTION
**Pull request by italoteix**
# Changes

## Related issues

Resolver #23 

## Added

- Timesheets collection to hold the status data
- New timesheets page table showing the status of several weeks and navigate directly to that referred week on click

## Removed

- Status field on records

## Changed

- Logic to save and fetch records keeping reference on timesheets

## How to test

- Check the full cycle of the application, create records, save them, approve, deny, change, delete, etc

## Screenshots

Screen Shot 2021-05-15 at 04 22 59

### CODE REVIEW

File: services/timesheets-service.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  }: GetTimesheetsProps) {
+    let query:
+      | firebase.default.firestore.CollectionReference<DocumentData>
+      | firebase.default.firestore.Query<
+          DocumentData
+        > = this.fire.firestore.collection("timesheets");
+
+    if (date) query = query.where("date", "==", date);
+
+    if (startDate)
```
Comment (sanderdejong88) [inline services/timesheets-service.ts:27]: you might want to guard against having both a date & startDate + endDate?

File: store/reports/actions.ts [hunk 2]
```diff
@@ -10,30 +10,54 @@ const actions: ActionTree<ReportsStoreState, RootStoreState> = {
...
       startDate,
       endDate,
     });
 
-    const travelRecords = await this.app.$travelRecordsService.getApprovedRecords(
-      {
-        startDate,
-        endDate,
-      }
+    const [
```
Comment (sanderdejong88) [inline store/reports/actions.ts:30]: We might want to look into this in the future. This is a lot of api calls for a single screen. Maybe we need to store more "meta" info timesheets. E.g. the user info or something.

Comment (sanderdejong88) [reply store/reports/actions.ts:30]: What do you think?

Comment (italoteix) [reply store/reports/actions.ts:30]: Agreed. This page is currently fetching almost every collection that we have on our DB.

**Pull request merged**